modoboa/admin/api/v2/tests.py

```diff
@@ -210,6 +208,25 @@ def test_create_with_bad_password(self):
         self.assertEqual(resp.status_code, 400)
         self.assertIn("password", resp.json())
 
+    def test_create_too_long_username(self):
+        """Test that it is not possible to create too long usernames (RFC5321)."""
+        url = reverse("v2:account-list")
+        toolong_username = "".join(["a" for i in range(66)])
```

Comment:
FYI this line can be also write (cost is lower)

`toolong_username = "a" * 66`

```
  0           0 RESUME                   0

  1           2 LOAD_CONST               0 ('aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')
              4 STORE_NAME               0 (toolong_username)
              6 RETURN_CONST             1 (None)
```
versus `toolong_username = "".join(["a" for i in range(66)])`

```
  0           0 RESUME                   0

  1           2 LOAD_CONST               0 ('')
              4 LOAD_ATTR                1 (NULL|self + join)
             24 PUSH_NULL
             26 LOAD_NAME                1 (range)
             28 LOAD_CONST               1 (66)
             30 CALL                     1
             38 GET_ITER
             40 LOAD_FAST_AND_CLEAR      0 (i)
             42 SWAP                     2
             44 BUILD_LIST               0
             46 SWAP                     2
        >>   48 FOR_ITER                 4 (to 60)
             52 STORE_FAST               0 (i)
             54 LOAD_CONST               2 ('a')
             56 LIST_APPEND              2
             58 JUMP_BACKWARD            6 (to 48)
        >>   60 END_FOR
             62 SWAP                     2
             64 STORE_FAST               0 (i)
             66 CALL                     1
             74 STORE_NAME               2 (toolong_username)
             76 RETURN_CONST             3 (None)
        >>   78 SWAP                     2
             80 POP_TOP
             82 SWAP                     2
             84 STORE_FAST               0 (i)
             86 RERAISE                  0
ExceptionTable:
  44 to 60 -> 78 [4]
```

Reply:
Thanks, indeed it was overcomplicated :). [Fixed](https://github.com/modoboa/modoboa/pull/3277/commits/4ec0a25ab5dad174655099e3067c16e8680f8172)